python/pyspark/sql/dataframe.py

```diff
@@ -1980,6 +1981,11 @@ def dropDuplicates(self, subset=None):
         |Alice|  5|    80|
         +-----+---+------+
         """
+        if subset is not None and not isinstance(subset, Iterable):
+            raise TypeError("Parameter 'subset' must be a list of columns")
+        if subset is not None and isinstance(subset, str):
+            raise TypeError("Parameter 'subset' must be a list of columns")
```

Comment:
oh, you changed. sorry for a back and forth. I think list and tuple check was correct - I checked the py4j conversion logic.

Reply:
I just tested that we could technically pass a generator

Reply:
👍

Reply:
```
>>> def generator():
...     yield "name"
... 
>>> df = spark.createDataFrame(
            [("Alice", 50), ("Alice", 60)],
            ["name", "age"]
        )
>>> df.dropDuplicates(generator()).collect()
[Row(name='Alice', age=50)]
```